integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java

```diff
@@ -293,27 +325,37 @@ public void deployWebAppViaREST(
    * @throws Exception
    */
   public void deployWebAppViaWLST(
-      String webappName, String webappLocation, String username, String password) throws Exception {
+      String webappName,
+      String webappLocation,
+      String appLocationInPod,
+      String username,
+      String password)
+      throws Exception {
+    String adminPod = domainUid + "-" + adminServerName;
 
-    TestUtils.kubectlcp(
-        webappLocation,
-        "/shared/applications/" + webappName + ".war",
-        domainUid + "-" + adminServerName,
-        domainNS);
-
-    TestUtils.kubectlcp(
-        projectRoot + "/integration-tests/src/test/resources/deploywebapp.py",
-        "/shared/deploywebapp.py",
-        domainUid + "-" + adminServerName,
-        domainNS);
-
-    TestUtils.kubectlcp(
-        projectRoot + "/integration-tests/src/test/resources/callpyscript.sh",
-        "/shared/callpyscript.sh",
-        domainUid + "-" + adminServerName,
-        domainNS);
-
-    callShellScriptByExecToPod(username, password, webappName);
+    String cmd =
+        " -- bash -c 'cat > " + appLocationInPod + "/" + webappName + ".war' < " + webappLocation;
+
+    TestUtils.kubectlexec(adminPod, domainNS, cmd);
+
+    TestUtils.kubectlexec(
+        adminPod,
+        domainNS,
+        " -- bash -c 'cat > "
+            + appLocationInPod
+            + "/deploywebapp.py' < "
+            + projectRoot
+            + "/integration-tests/src/test/resources/deploywebapp.py");
+    TestUtils.kubectlexec(
+        adminPod,
+        domainNS,
+        " -- bash -c 'cat > "
+            + appLocationInPod
+            + "/callpyscript.sh' < "
+            + projectRoot
+            + "/integration-tests/src/test/resources/callpyscript.sh");
+
```

Comment:
The same 'exec cat' pattern is used a few times in this change to cp files -- it'd be helpful to abstract this out via TestUtils.copyFileViaCat() or some-such function, just like was already done for kubectl cp via TestUtils.kubectlcp()...

Reply:
In theory, you should be able to simply replace the implementation of kubectlcp()....

Reply:
Ok. Will add a new method instead of replacing the implementation of kubectlcp() which could be confusing.